src/button/styled-components.js

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import {styled} from '../styles';
+import {KIND, SIZE, SHAPE} from './constants';
+
+export const BaseButton = styled(
+  'button',
+  ({$theme, $size, $kind, $shape}) => ({
+    ...$theme.typography[$size === SIZE.compact ? 'font200' : 'font300'],
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    border: 'none',
+    borderRadius: $shape === SHAPE.round ? '50%' : $theme.borders.radius200,
+    outline: 'none',
+    WebkitAppearance: 'none',
+    transitionProperty: 'background',
+    transitionDuration: $theme.animation.timing100,
+    transitionTimingFunction: $theme.animation.easeOutCurve,
+    cursor: 'pointer',
+    ':disabled': {
+      cursor: 'not-allowed',
+      backgroundColor: $theme.colors.mono300,
+      color: $theme.colors.mono600,
+    },
+    // Padding
+    ...($shape === SHAPE.round || $shape === SHAPE.square
+      ? {
+          paddingTop:
+            $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
+          paddingBottom:
+            $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
+          paddingLeft:
+            $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
+          paddingRight:
+            $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
+        }
+      : {
+          paddingTop:
+            $theme.sizing[$size === SIZE.compact ? 'scale200' : 'scale300'],
+          paddingBottom:
+            $theme.sizing[$size === SIZE.compact ? 'scale200' : 'scale300'],
+          paddingLeft: $theme.sizing.scale600,
+          paddingRight: $theme.sizing.scale600,
+        }),
+    // Kind style override
+    ...getStyleForKind({$theme, $kind}),
+  }),
+);
+
+export const EndEnhancer = styled('div', ({$theme}) => ({
+  display: 'flex',
```

Comment:
end element should dock to right always. Middle element can have `flex=auto` or there are other tricks
https://www.google.com/search?q=flex+last+element+right

Reply:
order - yes. But are you sure, that you will not have this case when right side will be of too much padding?
![image](https://user-images.githubusercontent.com/1851469/45234263-08f05800-b28a-11e8-9b10-f2500a25d92b.png)